app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java

```diff
@@ -1135,18 +1138,24 @@ WResult<WAllowOrDeny> onLoadRequest(@NonNull WSession aSession, @NonNull LoadReq
         Log.d(LOGTAG, "onLoadRequest: " + uri);
 
         if (aSession == mState.mSession) {
-            Log.d(LOGTAG, "Testing for UA override");
-
             String userAgentOverride = sUserAgentOverride.lookupOverride(uri);
 
-            // Here we set a default user agent if no override was found, BUT
-            // only if UA Mode is not Desktop, because the UA override
-            // takes precedence over mode overrides so if we set it, it will
-            // invalidate desktop mode UA. For VR and Mobile modes we don't change
-            // the UA so they are not affected.
-            if (mState.mSettings.getUserAgentMode() != WSessionSettings.USER_AGENT_MODE_DESKTOP &&
-                    userAgentOverride == null) {
-                userAgentOverride = DEFAULT_USER_AGENT;
+            Log.e(LOGTAG, "checking override: " + uri);
+
+            // Update the User-Agent according to the current settings, unless we are
+            // in Desktop mode (which sets its own user agent). These settings define:
+            //  - UA mode: "Mobile" or "Mobile VR"
+            //  - use Wolvic UA: report ourselves as "Firefox" (the GeckoView default) or "Wolvic"
+            int mode = mState.mSettings.getUserAgentMode();
+            if (mode != WSessionSettings.USER_AGENT_MODE_DESKTOP) {
+                int defaultMode = SettingsStore.getInstance(mContext).getUaMode();
+                if (mode != defaultMode) {
+                    setUaMode(defaultMode);
+                    mode = defaultMode;
+                }
```

Comment:
@svillar Done. I had to add some extra code so we don't reload all the tabs when the _"User-Agent mode"_ setting changes (because all of them get notified of the setting change).

The previous behaviour before all my changes was that you needed to open a new tab (i.e. create a new `Session`) to get the new UA value.